scripts/generate-zeus.sh

```diff
@@ -12,16 +13,34 @@ function generate() {
   local MV_PATH=$GEN_PATH/$1; shift
 
   # pass the rest of the arguments to zeus
-  (set -x; zeus $HASURA_GRAPHQL_ENDPOINT/v1/graphql $GEN_PATH --ts --rq $@)
-  test -d $GEN_PATH/zeus
-  test -d "$MV_PATH" && rm -r $MV_PATH
-  mv $GEN_PATH/zeus $MV_PATH
+  (set -x; zeus "$HASURA_GRAPHQL_ENDPOINT"/v1/graphql "$GEN_PATH" --ts --rq "$@")
+  test -d "$GEN_PATH"/zeus
+  test -d "$MV_PATH" && rm -r "$MV_PATH"
+  mv "$GEN_PATH"/zeus "$MV_PATH"
 
+
+
+  # shamelessly borrowed from
+  # https://gist.github.com/maxpoletaev/4ed25183427a2cd7e57a
+  case "$OSTYPE" in
+    darwin*)  PLATFORM="OSX" ;;
+    linux*)   PLATFORM="LINUX" ;;
+    bsd*)     PLATFORM="BSD" ;;
+    *)        PLATFORM="UNKNOWN" ;;
+  esac
   # mac sed needs sed -i '' -e
   # for other seds try removing -i '' -e
   # use `brew install gsed` on macos to get this
-  gsed -i 's,bigint"]:any,bigint"]:number,g' $MV_PATH/index.ts
-  gsed -i 's,bigint"]:unknown,bigint"]:number,g' $MV_PATH/index.ts
+  if [[ "$PLATFORM" == "OSX" || "$PLATFORM" == "BSD" ]]; then
+    sed -i "" 's,bigint"]:any,bigint"]:number,g' "$MV_PATH"/index.ts
+    sed -i "" 's,bigint"]:unknown,bigint"]:number,g' "$MV_PATH"/index.ts
+  elif [ "$PLATFORM" == "LINUX" ]; then
+    sed -i 's,bigint"]:any,bigint"]:number,g' "$MV_PATH"/index.ts
+    sed -i 's,bigint"]:unknown,bigint"]:number,g' "$MV_PATH"/index.ts
+  else
+    echo "unknown platform; exiting"
+    exit 1
+  fi
```

Comment:
@exrhizo let me know what you think about this

Reply:
You could get this from the output of `uname -a` (though this might break Windows? Not a Windows user)

Reply:
I don't think executing bash scripts in windows is a generally supported thing. As I understand it, it's okay to tell Windows devs to use [WSL](https://docs.microsoft.com/en-us/windows/wsl/about) and then do what's needed to support that environment.

Reply:
@levity does this script work without issue on your M1 macbook?

Reply:
I have a macbook m1 and it worked